analyzers/src/SonarAnalyzer.CSharp/Rules/ReturnEmptyCollectionInsteadOfNull.cs

```diff
@@ -23,131 +23,119 @@ namespace SonarAnalyzer.Rules.CSharp
                 }
             }
         }
 
         private static bool IsReturningCollection(SyntaxNodeAnalysisContext context)
         {
             var symbol = context.SemanticModel.GetDeclaredSymbol(context.Node);
-
             var methodSymbol = (symbol as IPropertySymbol)?.GetMethod ?? symbol as IMethodSymbol;
 
-            return methodSymbol != null &&
-                !methodSymbol.ReturnType.Is(KnownType.System_String) &&
-                !methodSymbol.ReturnType.DerivesFrom(KnownType.System_Xml_XmlNode) &&
-                methodSymbol.ReturnType.DerivesOrImplementsAny(CollectionTypes);
+            return methodSymbol != null
+                && !methodSymbol.ReturnType.Is(KnownType.System_String)
+                && !methodSymbol.ReturnType.DerivesFrom(KnownType.System_Xml_XmlNode)
+                && methodSymbol.ReturnType.DerivesOrImplementsAny(CollectionTypes);
         }
 
-        private static ArrowExpressionClauseSyntax GetExpressionBody(SyntaxNode node)
-        {
-            switch (node.Kind())
+        private static ArrowExpressionClauseSyntax GetExpressionBody(SyntaxNode node) =>
+            node switch
             {
-                case SyntaxKind.MethodDeclaration:
-                    return ((MethodDeclarationSyntax)node).ExpressionBody;
-
-                case SyntaxKind.PropertyDeclaration:
-                    var property = (PropertyDeclarationSyntax)node;
-
-                    if (property.ExpressionBody != null)
-                    {
-                        return property.ExpressionBody;
-                    }
-
-                    return property.AccessorList?.Accessors
-                        .FirstOrDefault(accessor => accessor.IsKind(SyntaxKind.GetAccessorDeclaration))
-                        ?.ExpressionBody();
+                MethodDeclarationSyntax method => method.ExpressionBody,
+                PropertyDeclarationSyntax property => property.ExpressionBody ?? GetGetAccessor(property)?.ExpressionBody(),
+                var _ when LocalFunctionStatementSyntaxWrapper.IsInstance(node) => ((LocalFunctionStatementSyntaxWrapper)node).ExpressionBody,
+                OperatorDeclarationSyntax @operator => @operator.ExpressionBody,
+                _ => null,
+            };
+
+        private static BlockSyntax GetBody(SyntaxNode node) =>
+            node switch
+            {
+                MethodDeclarationSyntax method => method.Body,
+                PropertyDeclarationSyntax property => GetGetAccessor(property)?.Body,
+                var _ when LocalFunctionStatementSyntaxWrapper.IsInstance(node) => ((LocalFunctionStatementSyntaxWrapper)node).Body,
+                OperatorDeclarationSyntax @operator => @operator.Body,
+                _ => null,
+            };
+
+        private static AccessorDeclarationSyntax GetGetAccessor(PropertyDeclarationSyntax property) =>
```

Comment:
The name of the method has 2 'Get' inside it (GetGetAccessor)

Reply:
Hey @csaba-sagi-sonarsource , do you mind the double Get? Here i am `get`ting a `Get` accessor, hence the get-get.

If you think it's not very clear, should i do something like the following?

- FetchGetAccessor
- ExtractGetAccessor
- RetrieveGetAccessor

Reply:
GetGet is indeed a bit weird. 
From my point of view, you can also just name it GetAccessor, as it returns the GetAccessor from the property. It is obvious it is "Getting you something".